ooniprobe/Model/Database/Measurement.mm

```diff
@@ -96,4 +96,44 @@ -(void)deleteObject{
     [self remove];
 }
 
+-(void)getExplorerUrl:(void (^)(NSString*))successcb onError:(void (^)(NSError*))errorcb {
+    NSURLComponents *components = [[NSURLComponents alloc] init];
+    components.scheme = @"https";
+    components.host = @"api.ooni.io";
+    components.path = @"/api/v1/measurements";
+    NSURLQueryItem *reportIdItem = [NSURLQueryItem queryItemWithName:@"report_id" value:self.report_id];
```

Comment:
What gives us confidence that `report_id` and `url_id` are not `nil`?

Reply:
Url can't be due to the db structure. We never call this function in non upload measurements, so report_id will alwayd be there.
To answer your question: if they are `nil` there will simply be a nil in the url, and the function will return non existent measurement_url

Reply:
> Url can't be due to the db structure. We never call this function in non upload measurements, so report_id will alwayd be there.

Thanks for the explanation. This is true _now_ but may change in the future. Especially with implicitly arguments that are class members. 🤔 

> To answer your question: if they are nil there will simply be a nil in the url, and the function will return non existent measurement_url

Right. I guess in such case the app would misbehave and we'd not clearly know why. I am quite sad that we're in a language that allows `nil`, but I guess this is it.

Are `report_id` and `url_id` externally settable? Maybe it makes sense to fail the test with a specific error in case they are not correctly set (i.e. they're `null`)?

Reply:
`report_id` is set only by the value returned by mk
`url_id` is an `Url` db object and must have a string value, so it can't be nil